GH-3001: default clientIds with application name

notizklotz opened the pull request:
Fixes: #GH-3001

Use Spring Boot's `spring.application.name` property as part of the default clientIds for Consumers, Producers and AdminClients. Helps with identifying problematic clients at server side.

* Only use as a fallback if clientId wasn't specified explicitly
* Do not use for Consumers with a specified groupId because KafkaConsumer will use the groupId as clientId which already is an identifiable default

sobychacko commented:
@notizklotz I am Starting to review the PR. There are some build failures due to checkstyle issues. See [here](https://github.com/spring-projects/spring-kafka/actions/runs/7960753296/job/22101177475?pr=3048). 
Also, you need to add some docs about these changes.

notizklotz commented:
> @notizklotz I am Starting to review the PR. There are some build failures due to checkstyle issues. See [here](https://github.com/spring-projects/spring-kafka/actions/runs/7960753296/job/22101177475?pr=3048). Also, you need to add some docs about these changes.

@sobychacko I fixed the checkstyle errors and added some docs to the "What’s new?" section

sobychacko commented:
@notizklotz Thanks. The problem with adding this in the `whats-new` doc is that this will be replaced for the next version (`3.3.0`). This deserves a separate small section inside the reference docs. Maybe add a new section for configuring `client.id`? One other thing - can you add your name as the `@author` in all the classes you modified and update any copyright years?

notizklotz commented:
@sobychacko Added section to reference docs, added author tag, checked copyright years, squashed commits and rebased to upstream main

sobychacko commented:
Thanks @notizklotz for the updates. One last comment - What do you think about extracting `spring.application.name` text literal into a common property since it is repeated in a handful of places?

notizklotz commented:
> Thanks @notizklotz for the updates. One last comment - What do you think about extracting `spring.application.name` text literal into a common property since it is repeated in a handful of places?

@sobychacko I considered that but decided against it, because I couldn't find an existing class where I think it would be a good fit. I didn't want to introduce a new class like "Constants" just for this literal. I also saw that the "client-id" literal is also copy/pasted across various classes and I thought to stick with the existing style.

sobychacko commented:
Fair enough. We can always refactor that later if need be.
